packages/beeq/src/components/tag/bq-tag.tsx

```diff
@@ -0,0 +1,230 @@
+import { Component, Element, Event, EventEmitter, h, Host, Method, Prop, State, Watch } from '@stencil/core';
+
+import { TAG_SIZE, TAG_VARIANT, TTagColor, TTagSize, TTagVariant } from './bq-tag.types';
+import { iconSize, textColor } from './helper';
+import { hasSlotContent, validatePropValue } from '../../shared/utils';
+
+/**
+ * @part wrapper - The wrapper container `<div>` of the element inside the shadow DOM.
+ * @part prefix - The `<span>` tag element that acts as prefix container (when icon exists in front of tag).
+ * @part text - The `<div>` element containing the text of the tag component.
+ */
+@Component({
+  tag: 'bq-tag',
+  styleUrl: './scss/bq-tag.scss',
+  shadow: {
+    delegatesFocus: true,
+  },
+})
+export class BqTag {
+  // Own Properties
+  // ====================
+
+  private prefixElem: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqTagElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() private hasPrefix = false;
+
+  // Public Property API
+  // ========================
+
+  /** The color style of the Tag */
+  @Prop({ reflect: true }) color: TTagColor;
+
+  /** The size of the Tag component */
+  @Prop({ reflect: true }) size: TTagSize = 'medium';
+
+  /** If true, the Tag component can be removed */
+  @Prop({ reflect: true }) removable: boolean = false;
+
+  /** If true, the Tag component will hidden (only if removable = `true`) */
+  @Prop({ reflect: true, mutable: true }) hidden: boolean;
+
+  /** The variant of Tag to apply on top of the variant */
+  @Prop({ reflect: true }) variant: TTagVariant = 'filled';
+
+  /** If true, the Tag will be disabled (only if clickable = `true`, no interaction allowed) */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /** If true, the Tag can be clickable */
+  @Prop({ reflect: true }) clickable: boolean = false;
+
+  /** If true, the Tag is selected (only if clickable = `true`) */
+  @Prop({ reflect: true, mutable: true }) selected: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('size')
+  @Watch('variant')
+  checkPropValues() {
+    validatePropValue(TAG_SIZE, 'medium', this.el, 'size');
+    validatePropValue(TAG_VARIANT, 'filled', this.el, 'variant');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Callback handler to be called when the tag is removable  */
+  @Event() bqHide: EventEmitter;
+
+  /** Callback handler to be called when the tag is not removable */
+  @Event() bqShow: EventEmitter;
+
+  /** Handler to be called when tag is clicked */
+  @Event() bqClick: EventEmitter<HTMLBqTagElement>;
+
+  /** Handler to be called when tag is focused */
+  @Event() bqFocus: EventEmitter<HTMLBqTagElement>;
```

Comment:
We should think about emitting a bqClose and bqOpen events for when the tag is hidden/shown